ci: phpstan + testmail

lpoujade opened the pull request.

lpoujade committed:
ci: add phpstan & testmail script

marien-probesys commented on .github/workflows/main.yml:
```diff
@@ -1 +1,52 @@
+
+jobs:
+  phpstan:
+    runs-on: ubuntu-latest
```
J'ai tendance à utiliser une version fixe de l’image et la mettre à jour à la main manuellement. Ça a tendance à péter lorsque `ubuntu-latest` change de version, et généralement c'est pas le bon moment pour corriger la CI.

```suggestion
    runs-on: ubuntu-24.04
```

marien-probesys commented on .github/workflows/main.yml:
```diff
@@ -1 +1,52 @@
+        run: ./vendor/bin/phpstan analyse src/
+
+  testmail:
+    runs-on: ubuntu-latest
```
Idem

```suggestion
    runs-on: ubuntu-24.04
```

marien-probesys commented on .github/workflows/main.yml:
```diff
@@ -1 +1,52 @@
+      - run: composer install --no-scripts --no-cache
+
+      - name: PHPStan Static Analysis
+        run: ./vendor/bin/phpstan analyse src/
```
Pas besoin de préciser `src/` puisqu’il est déjà présent dans le fichier `phpstan.neon` (comme ça on aura qu’un seul endroit à changer si on rajoute des dossiers à analyser).

Il faudrait aussi ajouter l’option `-v`, ça permet de connaître le nom de la règle qui échoue (pratique si on veut l’ignorer)

```suggestion
        run: ./vendor/bin/phpstan analyse -v
```

marien-probesys commented on .github/workflows/main.yml:
```diff
@@ -1 +1,52 @@
+          sed -i "s|SF_TOKEN_ENCRYPTION_SALT=.*|SF_TOKEN_ENCRYPTION_SALT=`openssl rand -base64 32`|" .env
+          sed -i "s|example.com|agentj.local|g" .env
+          sed -i "s|#UID=.*|UID=`id -u`|g" .env
+          sed -i "s|#GID=.*|GID=`id -g`|g" .env
```
Détail : on peut pas juste déclarer les variables d’environnement en mode : 

```yaml
env:
  COMPOSE_FILE: docker-compose.yml:compose.dev.yml:compose.test.yml
  DB_PASSWORD: strangeci
  DB_ROOT_PASSWORD: strangeci
  ...
```

Je suis pas sûr qu’on puisse exécuter les commandes comme `openssl` et compagnie, mais on peut utiliser ici des valeurs en dur, il n’y a pas d’enjeu de sécurité.

C’est juste que les `sed` ont tendance à me faire sourciller et que je trouve ça moins lisible.

Autre question : ils sortent d’où les identifiants de la base de données ? :thinking:

lpoujade replied:
Bonne question pour l'env, je test ça. Il y a juste pour les UID/GID dont on a vraiment besoin qu'on pourra pas le faire. Par contre le backticks fonctionnent bien sur le `run`

> Autre question : ils sortent d’où les identifiants de la base de données ? 🤔

De mon c… imagination sans limite 😁
C'est vrai qu'on peux laisser ceux par défault

lpoujade replied:
C'est peut être configurable mais pour l'instant je laisse la génération des clefs sinon gitguardian y trouve des secrets ([même là](https://github.com/Probesys/agentj/pull/86/commits/410def40372792f639704fbf0144878bfcb79d68))

D'ailleurs, ça nous permet de vérifier que les commandes qu'on proposent dans la doc fonctionnent bien

lpoujade replied:
Finalement je laisse comme ça pour l'instant pour être certain que les deux appels à docker aient bien le même env sans avoir besoin de dupliquer quoi que ce soit, je te laisse voir si tu merge comme ça ou si tu veux regarder s'il y a possibilité de mettre en env global et qu'il est bien pris en compte par docker

marien-probesys replied:
Ok ça marche, laissons ça comme ça pour l’instant :+1:

marien-probesys commented on .github/workflows/main.yml:
```diff
@@ -1 +1,52 @@
+            compose.dev.yml
+            compose.test.yml
+          up-flags: --build
+          down-flags: --volumes
```
Je savais pas qu’on pouvait faire ça !! :O Top !

lpoujade replied:
L'idéal serait que COMPOSE_FILE soit respecté quand même, pour l'instant ça default sur docker-compose.yml seul

marien-probesys commented on policyd-rate-limit/Dockerfile:
```diff
@@ -1,5 +1,5 @@
-FROM gitlab.probesys.com:5050/agentj/policyd-rate-limit
+FROM probesys38/policyd-rate-limit
```
Le changement n’est pas lié à la PR, si ?

lpoujade replied:
Oui et non, la ci ne fonctionne pas sans ce changement

marien-probesys replied:
Ok :+1: Perso j'aurais fait une PR séparée juste pour ce changement qui concerne aussi #87, mais c'est pas un problème non plus !

lpoujade committed:
Fix from reviews

- specify os version
- more details for phpstan
- try a better integration for env var

lpoujade committed:
Try to not trigger gitguardian secret detection

lpoujade committed:
Secrets stay secret

lpoujade committed:
Same env for everyone
